packages/gatsby-theme-store/src/sdk/product/useSku.ts

```diff
@@ -2,15 +2,39 @@ import { useCallback, useMemo } from 'react'
 
 import { useSearchParams } from '../state/useSearchParams'
 
-export interface Sku {
+export type Item = {
+  name: string
+  complementName?: string
   itemId: string
+  referenceId: Array<{ value: string }>
+  videos: Array<{ videoUrl: string }>
+  sellers: Array<{
+    sellerId: string
+    commercialOffer: {
+      availableQuantity: number
+      price: number
+    }
+  }>
+  variations?: Array<{ name: string; values: string[] }>
+  images: Array<{ imageUrl: string; imageText: string }>
 }
 
-interface Props<T extends Sku> {
-  items: T[]
+export type Sku = {
+  productId: string
+  productClusters: Array<{ name: string }>
+  productName: string
+  productReference: string
+  categoryTree: Array<{ name: string; href: string }>
+  description: string
+  linkText: string
+  specificationGroups: Array<{
+    name: string
+    specifications: Array<{ name: string; values: string[] }>
+  }>
+  items: Item[]
 }
 
-export const useSku = <T extends Sku>(product: Props<T>) => {
```

Comment:
Why can't we use generics in here ?

Reply:
@tlgimenes Why should we use? I don't know too much about their benefits, do you think it will be too limited without generics?

Reply:
Yes ! The thing cool about generics is that we don't need an object of type SKU for this function to work. All we need is. that this type have some properties. 

This is handy when working with fragments